collada/schema.py

```diff
@@ -14,21 +14,38 @@
 """This module contains helper classes and functions for working
 with the COLLADA 1.4.1 schema."""
 
+import os
 import lxml
 import lxml.etree
 from collada.util import bytes, BytesIO
 
-from pkg_resources import resource_string
+# the absolute directory of this file
+_cwd = os.path.abspath(os.path.expanduser(os.path.dirname(__file__)))
+
+
+def resource_string(file_name: str) -> str:
+    """
+    Get the value of a file in `collada/resources/{file_name}`
+    as a string.
+
+    Parameters
+    -----------
+    file_name
+      The name of the file in `collada/resources/{file_name}`
+
+    Returns
+    ----------
+    value
+      The contents of the file.
+    """
+    with open(os.path.join(_cwd, "resources", file_name)) as f:
```

Comment:
If I'm reading the docs correctly, isn't the replacement supposed to be to use `importlib.resources` instead of doing the path manipulation like this directly? Like would this work in all cases, e.g. if the module is being loaded from a zip archive or some non standard module loader?

https://importlib-resources.readthedocs.io/en/latest/migration.html#pkg-resources-resource-string

Reply:
Great question re: `pyz`, it probably doesn't. We could also just do the `sys.version_info >= (3, 8)` check and switch between `pkg_resources` and `importlib`. Here's the version-gated implementation of [resource_string](https://github.com/mikedh/pycollada/commit/3e2c24a6a81fadc3e2ff81462606cb830a632cf4) happy to switch this PR to that if preferable. 

For amusement I tried chasing down where it actually opens the file in `importlib`, my best guess is [here](https://github.com/python/cpython/blob/main/Lib/importlib/resources/abc.py#L85-L90). There's an awful lot of inheritance going on there but it seems like it probably handles the zip case 🤷. 

I usually just set `zip_safe=False` in the package definition and do the `_cwd` thing  but fine either way! Also another wrinkle is that `pkg_resources` is from setuptools, which nearly everyone has and the install requires, but is not part of the standard library haha.

Reply:
I dropped support for < 3.8 in the last release:
https://pycollada.readthedocs.io/en/latest/changelog.html#backwards-compatibility-notes

I've generally been following https://devguide.python.org/versions/ - looks like 3.8 actually just reached end of life and 3.13 was added, so I'll have to update.

But IIUC, importlib was added in 3.8, right? So we should be good to drop support for anything else?

Reply:
Ah looks like the feature we need is in 3.9 😆, here's 3.8 failing with just `importlib`:

```

    from collada import schema
collada/schema.py:43: in <module>
    COLLADA_SCHEMA_1_4_1 = _resource_string('schema-1.4.1.xml')
collada/schema.py:38: in _resource_string
    return resources.files("collada").joinpath("resources", file_name).read_text()
E   AttributeError: module 'importlib.resources' has no attribute 'files'
```

Reply:
Sorry, I know this has been a pain, but it looks like 3.8 did have methods but you have to use it slightly differently?

https://docs.python.org/3.8/library/importlib.html#module-importlib.resources

I think you want `importlib.resources.read_text`?

But I think to make it work, you might have to add an `__init__.py` in the resources subdir so that you can use `collada.resources` as the module to import from?

Reply:
hey Michael, did you plan to continue on this?

Reply:
Sorry for the delay, updated to use only `importlib`. 

I had to make `collada.resources` a submodule for this to work with the older `read_text` function, but that's probably not a bad idea anyway.